probe-rs/src/debug/registers.rs

```diff
@@ -174,24 +174,59 @@ impl DebugRegisters {
             .unwrap_or_else(|| "unknown register".to_string())
     }
 
+    /// Retrieve a refererence to a register by searching against an exact match of the [`RegisterRole`].
+    pub fn get_register_by_role(&self, register_role: &RegisterRole) -> Option<&DebugRegister> {
+        self.0.iter().find(|debug_register| {
```

Comment:
We should emit a warning message if there is more than two registers matching the role. Is this a valid case?

Reply:
I would call that situation a bug ... so maybe an error instead of a warning.

Reply:
If it is a bug, this should panic. We can do `filter()`, assert that it has `len()==1` and then get the sole entry.

Reply:
I'd rather not panic, because that causes the dap server to disappear under VSCode. 

This error will be propagated to the user, and the unwind will stop at the current frame.

Reply:
We discussed this before, the debugger should catch the panic and not expect an error that gives the impression it's recoverable, which it is not :)

Reply:
We can fix this later, but we definitely need to clean those up. They clutter the interface for no reason :)